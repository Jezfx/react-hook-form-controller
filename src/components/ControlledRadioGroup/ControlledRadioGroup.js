import React from "react";
import PropTypes from "prop-types";
import { Controller, useFormContext } from "react-hook-form";
import {
  RadioGroup,
  Radio,
  FormControlLabel,
  FormLabel,
  FormControl,
  FormHelperText,
} from "@material-ui/core";

const ControlledRadioGroup = ({ defaultValue, name, label, options = [] }) => {
  const { control, errors } = useFormContext();
  const error = errors[name];

  return (
    <FormControl error={!!error}>
      {label && <FormLabel>{label}</FormLabel>}
      <Controller
        name={name}
        control={control}
        defaultValue={false}
        render={({ onChange, value = "", name }) => (
          <RadioGroup
            key={name}
            name={name}
            defaultValue={defaultValue}
            value={value}
            onChange={onChange}
          >
            {options?.map(({ ...props }) => (
              <FormControlLabel
                key={props.name}
                control={<Radio />}
                {...props}
              />
            ))}
          </RadioGroup>
        )}
      />
      <FormHelperText>{error?.message}</FormHelperText>
    </FormControl>
  );
};

ControlledRadioGroup.propTypes = {
  options: PropTypes.arrayOf(PropTypes.object),
  name: PropTypes.string,
};

ControlledRadioGroup.defaultProps = {
  options: [],
};

export default ControlledRadioGroup;
