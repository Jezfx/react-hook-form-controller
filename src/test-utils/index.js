import { render } from "@testing-library/react";
import { useForm, FormProvider } from "react-hook-form";

export const renderWithProviders = (ui, { defaultValues = {} } = {}) => {
  const Wrapper = ({ children }) => {
    const methods = useForm({});
    return <FormProvider {...methods}>{children}</FormProvider>;
  };
  return render(ui, { wrapper: Wrapper });
};
