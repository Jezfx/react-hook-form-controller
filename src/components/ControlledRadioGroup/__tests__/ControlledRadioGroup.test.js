import React from "react";
import ControlledRadioGroup from "../";
import { renderWithProviders } from "../../../test-utils";

describe("ControlledRadioGroup", () => {
  it("should match snapshot", () => {
    const { asFragment } = renderWithProviders(
      <ControlledRadioGroup options={[]} name="foo" label="Foo" />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
