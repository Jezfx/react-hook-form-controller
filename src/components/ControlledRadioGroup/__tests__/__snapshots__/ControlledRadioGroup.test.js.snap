// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlledRadioGroup should match snapshot 1`] = `
<DocumentFragment>
  <fieldset
    class="MuiFormControl-root"
  >
    <label
      class="MuiFormLabel-root"
    >
      Foo
    </label>
    <div
      class="MuiFormGroup-root"
      role="radiogroup"
    />
    <p
      class="MuiFormHelperText-root"
    />
  </fieldset>
</DocumentFragment>
`;
